pandas/tests/test_index.py

```diff
@@ -60,6 +60,16 @@ def test_shift(self):
         self.assertRaises(NotImplementedError, idx.shift, 1)
         self.assertRaises(NotImplementedError, idx.shift, 1, 2)
 
+    def test_create_index_existing_name(self):
+
+        # GH11193, when an existing index is passed, and a new name is not specified, the new index should inherit the
+        # previous object name
+        idx = self.create_index()
+        idx.name = "foo"
+        result = pd.Index(idx)
+        self.assertTrue("'foo'" in str(result))
```

Comment:
What about the case when the object passed to `Index` is a `MultiIndex`? I think this is a corner case and should be avoided in principle. To get over this, I would write the unit test as follows, even though is not very elegant:

        expected = self.create_index()
        if not isinstance(expected, MultiIndex):
            expected.name = 'foo'
            result = pd.Index(expected)
            tm.assert_index_equal(result, expected)

            result = pd.Index(expected, name='bar')
            expected.name = 'bar'
            tm.assert_index_equal(result, expected)

Should `Index` raise an error when we try to pass a `MultiIndex` object?